---
title: 'Özel Durumlar: failwith İşlevi (F#)'
description: "Nasıl 'failwith' işlevi bir F # özel durum oluşturur öğrenin."
ms.date: 05/16/2016
ms.openlocfilehash: 69a2eb69e0157d3bde8cb8884cb0ae960634dddc
ms.sourcegitcommit: db8b83057d052c1f9f249d128b08d4423af0f7c2
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 11/02/2018
ms.locfileid: "43863435"
---
# <a name="exceptions-the-failwith-function"></a><span data-ttu-id="e3444-103">Özel Durumlar: failwith İşlevi</span><span class="sxs-lookup"><span data-stu-id="e3444-103">Exceptions: The failwith Function</span></span>

<span data-ttu-id="e3444-104">`failwith` İşlevi bir F # özel durum oluşturur.</span><span class="sxs-lookup"><span data-stu-id="e3444-104">The `failwith` function generates an F# exception.</span></span>

## <a name="syntax"></a><span data-ttu-id="e3444-105">Sözdizimi</span><span class="sxs-lookup"><span data-stu-id="e3444-105">Syntax</span></span>

```fsharp
failwith error-message-string
```

## <a name="remarks"></a><span data-ttu-id="e3444-106">Açıklamalar</span><span class="sxs-lookup"><span data-stu-id="e3444-106">Remarks</span></span>

<span data-ttu-id="e3444-107">*Hata iletisi dizesi* önceki sözdiziminde bir değişmez değer dize türünde bir değer mi `string`.</span><span class="sxs-lookup"><span data-stu-id="e3444-107">The *error-message-string* in the previous syntax is a literal string or a value of type `string`.</span></span> <span data-ttu-id="e3444-108">Bu duruma `Message` özel durumun özelliği.</span><span class="sxs-lookup"><span data-stu-id="e3444-108">It becomes the `Message` property of the exception.</span></span>

<span data-ttu-id="e3444-109">Tarafından oluşturulan özel durum `failwith` olduğu bir `System.Exception` adına sahip bir başvurudur özel durum `Failure` F # kodu.</span><span class="sxs-lookup"><span data-stu-id="e3444-109">The exception that is generated by `failwith` is a `System.Exception` exception, which is a reference that has the name `Failure` in F# code.</span></span> <span data-ttu-id="e3444-110">Aşağıdaki kod kullanışını `failwith` özel durum oluşturabilir.</span><span class="sxs-lookup"><span data-stu-id="e3444-110">The following code illustrates the use of `failwith` to throw an exception.</span></span>

[!code-fsharp[Main](../../../../samples/snippets/fsharp/lang-ref-2/snippet6001.fs)]

## <a name="see-also"></a><span data-ttu-id="e3444-111">Ayrıca bkz.</span><span class="sxs-lookup"><span data-stu-id="e3444-111">See also</span></span>

- [<span data-ttu-id="e3444-112">Özel Durum İşleme</span><span class="sxs-lookup"><span data-stu-id="e3444-112">Exception Handling</span></span>](index.md)
- [<span data-ttu-id="e3444-113">Özel Durum Türleri</span><span class="sxs-lookup"><span data-stu-id="e3444-113">Exception Types</span></span>](exception-types.md)
- [<span data-ttu-id="e3444-114">Özel durumlar: `try...with` ifadesi</span><span class="sxs-lookup"><span data-stu-id="e3444-114">Exceptions: The `try...with` Expression</span></span>](the-try-with-expression.md)
- [<span data-ttu-id="e3444-115">Özel durumlar: `try...finally` ifadesi</span><span class="sxs-lookup"><span data-stu-id="e3444-115">Exceptions: The `try...finally` Expression</span></span>](the-try-finally-expression.md)
- [<span data-ttu-id="e3444-116">Özel durumlar: `raise` işlevi</span><span class="sxs-lookup"><span data-stu-id="e3444-116">Exceptions: the `raise` Function</span></span>](the-raise-function.md)
